host_source.go

```diff
@@ -497,12 +497,24 @@ func (s *Session) hostInfoFromMap(row map[string]interface{}, host *HostInfo) (*
 				return nil, fmt.Errorf(assertErrorMsg, "rpc_address")
 			}
 			host.rpcAddress = net.ParseIP(ip)
+		case "native_address":
+			ip, ok := value.(string)
+			if !ok {
+				return nil, fmt.Errorf(assertErrorMsg, "native_address")
+			}
+			host.rpcAddress = net.ParseIP(ip)
 		case "listen_address":
 			ip, ok := value.(string)
 			if !ok {
 				return nil, fmt.Errorf(assertErrorMsg, "listen_address")
 			}
 			host.listenAddress = net.ParseIP(ip)
+		case "native_port":
+			native_port, ok := value.(int)
+			if !ok {
+				return nil, fmt.Errorf(assertErrorMsg, "native_port")
+			}
+			host.port = native_port
```

Comment:
It seems we should store the native port into a new field, there are different ports fields in system.local: `broadcast_port`, `listen_port`, `rpc_port` and in system.peers_v2: `peer_port`, `native_port`, `preferred_port`.
So it should be clear from the API which port we return, especially since we already have methods like BroadcastAddress/ListenAddress/RPCAddress/PreferredIP. We should probably add corresponding methods that return both address and port and deprecate the methods that return only address, for example:

```go
func (h *HostInfo) BroadcastAddressPort() (net.IP, int) {
	h.mu.RLock()
	defer h.mu.RUnlock()
	return h.broadcastAddress, h.broadcastPort
}
```

Is there some documentation on how `broadcast_port`, `listen_port`, `rpc_port` correspond to `peer_port`, `native_port`, `preferred_port`? I couldn't find any Cassandra docs on this. Is rpc_port the same as native_port?

Reply:
`broadcast_port` is by default `7000` and it corresponds to `storage_port` in `cassandra.yaml`. That is port via Cassandra nodes talk to each other internaly. `listen_port` is same, 7000. If you are not using `broadcast_address` (it is commented out, then it defaults to `listen_address`.

In times of pre-4.0, `rpc_port` in cassandra.yaml configuration was meant to represent Thrift port. Whole Thrift was abandoned / removed from 4.0 and afaik it seems like it is same as transport  / cql port.

peer_port is by default 7000 and that is used for internal communication between nodes, basically same as `storage_port` in the configration.

I am not sure about preferred port, that is not important in this context.

You see it is quite wild, I would just keep this as simple as possible.

Reply:
Thanks for the explanation!